src/main/java/com/gildedrose/RedWine.java

```diff
@@ -0,0 +1,39 @@
+package com.gildedrose;
+
+public class RedWine extends DefaultItem {
+
+    private final float redWineQualityDegradationPastSellIn;
+    private final float redWineQualityDegradation;
+    private final Item item;
+    private float preciseQuality;
+
+    public RedWine(Item item) {
+        super(item);
+        this.redWineQualityDegradation = super.qualityDegradation/3.0f;
+        this.redWineQualityDegradationPastSellIn = this.redWineQualityDegradation * 2.0f;
+        this.item = item;
+        this.preciseQuality = item.quality;
+    }
+
+    public void age() {
+        item.quality = updateQuality();
+        item.sellIn = updateSellIn();
+    }
+
+    private int updateQuality() {
+        if(item.sellIn <= 0) {
```

Comment:
**Suggestion**: Possible to reduce the amount of `if... else...` statements w/ ternaries or something else more concise?

Reply:
This is a good question cause I am unsure about ternaries. They definitely make the code more concise but I feel like at a certain point they become more of a pain to read and more showing off that you can do some nifty hack. 

I'd appreciate input from anyone who wants to on this subject. I'm asking because I want to get ahead of this early in my career and hear different opinions on when to use them and when to avoid them.